lib/modules/dosomething/dosomething_payment/dosomething_payment.module

```diff
@@ -214,7 +227,7 @@ function dosomething_payment_form_submit($form, &$form_state) {
   // Expected amount to charge to Stripe should be in cents.
   $values['amount'] = $values['amount'] * 100;
   // Send Payment request to Stripe API.
-  if (dosomething_payment_post_payment_stripe($values)) {
+  if (dosomething_payment_post_payment_stripe($values) && !isset($values['email'])) {
```

Comment:
Oh right, because we're calling that function and checking its return value. Yeah Ill move it up